### ISSUE TITLE
MosaicService is broken

### ISSUE THREAD

**Issue by dgarcia360** [nemtech/nem2-sdk-typescript-javascript#82]
1. Expected behavior and actual behavior.

MosaicService does not work properly since the latest cow update.

Error: 
>TypeError: Cannot read property 'length' of undefined
    at new UInt64 (<...>\node_modules\nem2-sdk\src\model\UInt64.ts:61:23)
    at mosaicInfosDTO.map (<...>\node_modules\nem2-sdk\src\infrastructure\MosaicHttp.ts:100:25)
    at Array.map (<anonymous>)
    at MapSubscriber.rxjs_1.from.pipe.operators_1.map [as project] (<...>\project\node_modules\nem2-sdk\src\infrastructure\MosaicHttp.ts:94:39)
    at MapSubscriber._next (<...>\project\node_modules\rxjs\src\internal\operators\map.ts:81:29)
    at MapSubscriber.Subscriber.next (<...>\project\node_modules\rxjs\src\internal\Subscriber.ts:104:12)
    at <...>\node_modules\rxjs\src\internal\util\subscribeToPromise.ts:8:20
    at process._tickCallback (internal/process/next_tick.js:68:7)

2. Steps to reproduce the problem.

Code executed:

```
import {AccountHttp, Address, MosaicHttp, MosaicService, NamespaceHttp} from 'nem2-sdk';

const url = 'http://localhost:3000'; // replace with URL
const accountHttp = new AccountHttp(url);
const mosaicHttp = new MosaicHttp(url);
const namespaceHttp = new NamespaceHttp(url);
const mosaicService = new MosaicService(accountHttp, mosaicHttp, namespaceHttp);

const address = Address.createFromRawAddress("SDSMQKMKCAE3LHGKTDNE7NYJOYEFDKLAWAKWKRAM");

mosaicService
    .mosaicsAmountViewFromAddress(address)
    .subscribe(mosaics => console.log(mosaics),
        err => console.error(err));
```

Additional comments: Evaluate if ``mosaicService`` constructor needs to pass ``namespaceHttp`` or if this parameter can be omitted.

**Comment by evias** [comment]
Merged in PR #84

**Comment by hungtranqn** [comment]
Hello, Now I try to run this example but it still broken.
Error:
```
TypeError: Cannot read property 'length' of undefined
    at new UInt64 (/home/hungtd/seft-learn/express/node_modules/nem2-sdk/dist/src/model/UInt64.js:48:23)
    at mosaicInfosDTO.map (/home/hungtd/seft-learn/express/node_modules/nem2-sdk/dist/src/infrastructure/MosaicHttp.js:69:394)
    at Array.map (<anonymous>)
    at MapSubscriber.rxjs_1.from.pipe.operators_1.map [as project] (/home/hungtd/seft-learn/express/node_modules/nem2-sdk/dist/src/infrastructure/MosaicHttp.js:68:35)
    at MapSubscriber._next (/home/hungtd/seft-learn/express/node_modules/rxjs/internal/operators/map.js:49:35)
    at MapSubscriber.Subscriber.next (/home/hungtd/seft-learn/express/node_modules/rxjs/internal/Subscriber.js:66:18)
    at /home/hungtd/seft-learn/express/node_modules/rxjs/internal/util/subscribeToPromise.js:7:24
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
code:
```
var nem = require('nem2-sdk');
var operators = require('rxjs/operators')
let url = 'http://40.90.163.184:3000';

const accountHttp = new nem.AccountHttp(url);
const mosaicHttp = new nem.MosaicHttp(url);
const mosaicService = new nem.MosaicService(accountHttp, mosaicHttp);
const address = nem.Address.createFromRawAddress("SCA7ZS-2B7DEE-BGU3TH-SILYHC-RUR32Y-YE55ZB-LYA2");
mosaicService
    .mosaicsAmountViewFromAddress(address)
    .pipe(
        operators.mergeMap((_) => _)
    )
    .subscribe(mosaic => console.log('You have', mosaic.relativeAmount(), mosaic.fullName()),
        err => console.error(err));
```
package.json
```
...
    "nem2-sdk": "^0.12.1",
    "rxjs": "^6.5.2"
...
```
Thanks,

**Comment by evias** [comment]
@rg911 can you try to reproduce please ?

**Comment by hungtranqn** [comment]
Hi I see you update code but when I run again my code example above I got same error,

I thinks my investigate will help you fix this bug faster: 
1. in file : https://github.com/nemtech/nem2-sdk-typescript-javascript/blob/master/src/infrastructure/MosaicHttp.ts method getMosaics 
   - what happen if 
``` 
let mosaicFlag;
let divisibility;
let duration;
``` 
do not pass value.
2. So I investigated in nem2-library (before you update code) and I find root cause:  if you do it longtime you can understand :). (this is one object I was log, all of them have same "properties" 
```
MosaicDefinitionDTO {
  mosaicId: [ 3178952541, 1005288441 ],
  supply: [ 100, 0 ],
  height: [ 406187, 0 ],
  owner:
   'AC1A6E1D8DE5B17D2C6B1293F1CAD3829EEACF38D09311BB3C8E5A880092DE26',
  revision: 1,
  properties:              <-------------------------------- "properties" 
   [ MosaicPropertyDTO { id: undefined, value: undefined },
     MosaicPropertyDTO { id: undefined, value: undefined },
     MosaicPropertyDTO { id: undefined, value: undefined } ] }
```
   - properties: always is array null MosaicPropertyDTO object.

Thank you,

**Comment by rg911** [comment]
I am not quite understand what are you asking for. Do you mean when create a mosaic with null property the MosaicHttp throw exception?

**Comment by hungtranqn** [comment]
Hi, you can check this file: https://github.com/nemtech/nem2-sdk-typescript-javascript/blob/master/src/infrastructure/MosaicHttp.ts  the metthod

```
public getMosaics(mosaicIds: MosaicId[]): Observable < MosaicInfo[] > {
const mosaicIdsBody = {
mosaicIds: mosaicIds.map((id) => id.toHex()),
};
return this.getNetworkTypeObservable().pipe(
mergeMap((networkType) => observableFrom(
this.mosaicRoutesApi.getMosaics(mosaicIdsBody)).pipe(map((mosaicInfosDTO: MosaicInfoDTO[]) => {
return mosaicInfosDTO.map((mosaicInfoDTO) => {
let mosaicFlag;
let divisibility;
let duration;
if (mosaicInfoDTO.mosaic.properties[MosaicPropertyType.MosaicFlags].value) {
mosaicFlag = mosaicInfoDTO.mosaic.properties[MosaicPropertyType.MosaicFlags].value;
}
if (mosaicInfoDTO.mosaic.properties[MosaicPropertyType.Divisibility].value) {
divisibility = mosaicInfoDTO.mosaic.properties[MosaicPropertyType.Divisibility].value;
}
if (mosaicInfoDTO.mosaic.properties[MosaicPropertyType.Duration].value) {
duration = mosaicInfoDTO.mosaic.properties[MosaicPropertyType.Divisibility].value;
}
return new MosaicInfo(
mosaicInfoDTO.meta.id,
new MosaicId(mosaicInfoDTO.mosaic.mosaicId),
new UInt64(mosaicInfoDTO.mosaic.supply),
new UInt64(mosaicInfoDTO.mosaic.height),
PublicAccount.createFromPublicKey(mosaicInfoDTO.mosaic.owner, networkType),
mosaicInfoDTO.mosaic.revision,
new MosaicProperties(
    new UInt64(mosaicFlag),
    (new UInt64(divisibility)).compact(),
    duration ? new UInt64(duration) : undefined,
),
);
});
}))));
}
```
focus line: `new MosaicProperties(
    new UInt64(mosaicFlag),
    (new UInt64(divisibility)).compact(),
    duration ? new UInt64(duration) : undefined,
)`
and check the contructor of UInt64 : https://github.com/nemtech/nem2-sdk-typescript-javascript/blob/master/src/model/UInt64.ts
```
constructor(uintArray: number[]) {
        if (uintArray.length !== 2 || uintArray[0] < 0 || uintArray[1] < 0) {
            throw new Error('uintArray must be be an array of two uint numbers');
        }
        this.lower = uintArray[0];
        this.higher = uintArray[1];
    }
```
uintArray.length --> this throw exception

**Comment by evias** [comment]
fixed in 0.12.3
